src/Service.Tests/OpenApiDocumentor/ParameterValidationTests.cs

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.OpenApi.Models;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Azure.DataApiBuilder.Service.Tests.OpenApiIntegration;
+
+/// <summary>
+/// Integration tests validating correct path and query parameters are created
+/// for different entities in DAB's REST API endpoint.
+/// </summary>
+[TestCategory(TestCategory.MSSQL)]
+[TestClass]
+public class ParameterValidationTests
+{
+    private const string CUSTOM_CONFIG = "parameter-config.MsSql.json";
+    private const string MSSQL_ENVIRONMENT = TestCategory.MSSQL;
+
+    /// <summary>
+    /// Validates the path parameters are correctly generated for GET methods.
+    /// </summary>
+    /// <param name="entityName">The name of the entity.</param>
+    /// <param name="objectName">The name of the database object.</param>
+    /// <param name="entitySourceType">The source type of the entity.</param>
+    /// <returns></returns>
+    [DataTestMethod]
+    [DataRow("BooksTable", "books", EntitySourceType.Table, DisplayName = "Table with path parameter id")]
+    [DataRow("BooksView", "books_view_all", EntitySourceType.View, DisplayName = "View with path parameter id")]
+    public async Task ValidatePathParametersForTablesAndViews(string entityName, string objectName, EntitySourceType entitySourceType)
+    {
+        EntitySource entitySource = new(Object: objectName, entitySourceType, null, null);
+        OpenApiDocument openApiDocument = await GenerateOpenApiDocumentForGivenEntity(entityName, entitySource);
+
+        foreach ((string pathName, OpenApiPathItem pathItem) in openApiDocument.Paths)
+        {
+            string apiPathWithParam = $"/{entityName}/id/{{id}}";
+            if (pathName.Equals(apiPathWithParam))
+            {
+                Assert.IsTrue(pathItem.Parameters.Count is 1);
+                OpenApiParameter pathParameter = pathItem.Parameters.First();
+                Assert.AreEqual("id", pathParameter.Name);
+                Assert.AreEqual(ParameterLocation.Path, pathParameter.In);
+                Assert.IsTrue(pathParameter.Required);
+            }
+            else
+            {
+                // Get All method with path /entityName, will have no path parameters.
+                Assert.IsTrue(pathItem.Parameters.Count is 0);
```

Comment:
woudldn't `POST` hit this condition too? POST requests hit `/entityName` paths.

Reply:
right, POST will also hit this path